packages/vaadin-crud/src/vaadin-dialog-layout.js

```diff
@@ -161,33 +166,42 @@ class DialogLayout extends ElementMixin(ThemableMixin(PolymerElement)) {
     this._ensureChildren();
   }
 
+  __moveChildNodes(target) {
+    // Teleport header node
+    target.appendChild(this.__header);
+
+    // For custom form, remove slot attribute
+    // so that it is properly moved to dialog.
+    if (this.form.hasAttribute('slot')) {
+      this.form.removeAttribute('slot');
+    }
+
+    // Teleport edit form
+    target.appendChild(this.form);
+
+    // Teleport footer nodes
+    this.__footer.forEach((node) => target.appendChild(node));
+
+    // Wait to set label until slotted element has been moved.
+    setTimeout(() => {
+      this.__ariaLabel = this.__header.textContent.trim();
+    });
```

Comment:
This part is updated to avoid additional `querySelectorAll`. I'm not sure if `setTimeout` is needed here.
The original implementation has been since https://github.com/vaadin/vaadin-crud/commit/1caff128384def117a6f178ffd4a0121d0a43f42.